src/main/java/uk/gov/companieshouse/pscstatementdataapi/services/PscStatementService.java

```diff
@@ -31,4 +45,49 @@ private PscStatementDocument getPscStatementDocument(String companyNumber, Strin
             new ResourceNotFoundException(HttpStatus.NOT_FOUND, String.format(
                     "Resource not found for statement ID: %s, and company number: %s", statementId, companyNumber)));
   }
+
+  public void processPscStatement(String contextId, String companyNumber, String statementId,
+                                  CompanyPscStatement companyPscStatement) {
+
+    boolean isLatestRecord = isLatestRecord(companyNumber, statementId, companyPscStatement.getDeltaAt());
+
+    if (isLatestRecord) {
+
+      PscStatementDocument document = pscStatementTransformer.transformPscStatement(companyNumber, statementId, companyPscStatement);
+
+      saveToDb(contextId, companyNumber, statementId, document);
+    } else {
+      logger.info("Psc Statement not persisted as the record provided is not the latest record.");
+    }
+  }
+
+  private boolean isLatestRecord(String companyNumber, String statementId, String deltaAt) {
+    List<PscStatementDocument> statements = pscStatementRepository.findUpdatedPscStatement(companyNumber, statementId, dateTransformer.transformDate(deltaAt));
+    return statements.isEmpty();
+  }
+
+  private void saveToDb(String contextId, String companyNumber, String statementId, PscStatementDocument document) {
+
+    Created created = getCreatedFromCurrentRecord(companyNumber, statementId);
+    if(created == null) {
+      document.setCreated(new Created().setAt(document.getUpdated().getAt()));
+    } else {
+      document.setCreated(created);
+    }
+
+    try {
+      pscStatementRepository.save(document);
+      logger.info(String.format("Psc statement is updated in MongoDb for context id: %s, company number: %s, and statement id: %s",
+              contextId, companyNumber, statementId));
+    } catch (IllegalArgumentException illegalArgumentEx) {
+      throw new BadRequestException("Saving to MongoDb failed", illegalArgumentEx);
+    }
+
+  }
+
+  private Created getCreatedFromCurrentRecord(String companyNumber,String statementId) {
+    Optional<PscStatementDocument> document = pscStatementRepository.getPscStatementByCompanyNumberAndStatementId(companyNumber, statementId);
+    return document.isPresent() ? document.get().getCreated(): null;
```

Comment:
orElse() would maybe be a bit cleaner? Not a strong preference though

Reply:
The problem I'm running into is that getCreated is not part of Optional so .orElse(null) will still call getCreated on null and throw a null pointer. I can get the code working with the below but am not a fan of creating a new document just to avoid a null pointer.

`return document.orElse(new PscStatementDocument()).getCreated();`